### ISSUE TITLE
Reindex API support for reindex-in-place

### ISSUE THREAD

**Issue by rpedela** [elastic/elasticsearch#27164]
I would like to see support for reindex-in-place added to the Reindex API. The primary use case for me is changing a field's analysis but the source is the same. For example, I often need to update index time synonyms.

The primary advantage over creating a new index is reducing operational complexity. While creating a new index is pretty simple, the size of the index can cause problems. You effectively have to have double the storage which can be cost prohibitive for large indices, and it can take a very long time to reindex. For very large indices, ```_reindex``` may require adding more servers to the cluster which can create more complexity (and require more money) even though the ES clustering code is quite robust.

In other words, most of my operational headaches are the result of needing to reindex large indices. The reindex API alone was a significant improvement over using ```_scroll```, and I think adding in-place support would reduce my operational headaches further. I am sure an in-place option would have to be more restrictive in functionality, but I personally would be okay with that.

I am imagining internally performing ```_scroll``` and ```_update``` rather than having to do it myself.

**Comment by rpedela** [comment]
Nevermind. Update by query does what I need.
